server/src/main/java/org/opensearch/discovery/PeerFinder.java

```diff
@@ -128,6 +139,23 @@ public PeerFinder(
         );
     }
 
+    public synchronized void setFindPeersInterval(boolean localNodeCommissioned) {
+        this.localNodeCommissioned = localNodeCommissioned;
+        findPeersInterval = localNodeCommissioned
+            ? DISCOVERY_FIND_PEERS_INTERVAL_SETTING.get(settings)
+            : DISCOVERY_FIND_PEERS_INTERVAL_DURING_DECOMMISSION_SETTING.get(settings);
+        logger.info(
+            "setting findPeersInterval to [{}] as node commission status = [{}] for local node [{}]",
+            findPeersInterval,
+            localNodeCommissioned,
+            transportService.getLocalNode()
+        );
+    }
+
+    public boolean localNodeCommissioned() {
+        return localNodeCommissioned;
+    }
+
```

Comment:
PeerFinder should not be the source of truth, move this to coordinator instead

Reply:
Sure moving this flag to Coordinator